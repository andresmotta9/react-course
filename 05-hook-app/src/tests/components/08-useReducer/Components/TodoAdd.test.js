import React from 'react';
import {shallow} from 'enzyme';
import { TodoAdd } from '../../../../components/08-useReducer/components/TodoAdd';

describe('Tests on <TodoAdd />', () => {

    const hanldeAddTodo = jest.fn();

    const wrapper = shallow( 
        <TodoAdd
            hanldeAddTodo={ hanldeAddTodo}
        /> 
    )

    test('should show correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });


    test('should not call the handleAddTodo', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });
        
        expect( hanldeAddTodo ).toHaveBeenCalledTimes(0);
    });
    
    test('should call the hanldeAddTodo function', () => {
        
        const value = 'Learn Express';
        wrapper.find('input').simulate('change', { 
            target: { 
                value,
                name: 'description', 
            }}
        );
        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault(){} });
        
        expect( hanldeAddTodo ).toHaveBeenCalledTimes(1);
        expect( hanldeAddTodo ).toHaveBeenCalledWith( expect.any(Object) );
        expect( hanldeAddTodo ).toHaveBeenCalledWith( {
            id: expect.any(Number),
            desc: value,
            done: false,
         } );

        expect( wrapper.find('input').prop('value') ).toBe('');
    });
    
    
})
