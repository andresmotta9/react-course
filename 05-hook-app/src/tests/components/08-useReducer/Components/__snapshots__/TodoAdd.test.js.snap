// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <TodoAdd /> should show correctly 1`] = `
<Fragment>
  <h4>
    Add TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary btn-block mt-1"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
